diff --git a/node_modules/@remix-run/react/components.d.ts b/node_modules/@remix-run/react/components.d.ts
index d23e528..210ad01 100644
--- a/node_modules/@remix-run/react/components.d.ts
+++ b/node_modules/@remix-run/react/components.d.ts
@@ -90,7 +90,7 @@ export declare function PrefetchPageLinks({ page, ...dataLinkProps }: PrefetchPa
  * @see https://remix.run/api/remix#meta-links-scripts
  */
 export declare function Meta(): JSX.Element;
-declare type ScriptProps = Omit<React.HTMLProps<HTMLScriptElement>, "children" | "async" | "defer" | "src" | "type" | "noModule" | "dangerouslySetInnerHTML" | "suppressHydrationWarning">;
+declare type ScriptProps = { vanilla?: boolean } & Omit<React.HTMLProps<HTMLScriptElement>, "children" | "async" | "defer" | "src" | "type" | "noModule" | "dangerouslySetInnerHTML" | "suppressHydrationWarning">;
 /**
  * Renders the `<script>` tags needed for the initial render. Bundles for
  * additional routes are loaded later as needed.
diff --git a/node_modules/@remix-run/react/components.js b/node_modules/@remix-run/react/components.js
index 8f86eda..7b6654f 100644
--- a/node_modules/@remix-run/react/components.js
+++ b/node_modules/@remix-run/react/components.js
@@ -603,11 +603,21 @@ function Scripts(props) {
     matches,
     pendingLocation,
     clientRoutes,
-    serverHandoffString
+    serverHandoffString,
   } = useRemixEntryContext();
   React__namespace.useEffect(() => {
     isHydrated = true;
   }, []);
+
+  let {vanilla, ..._props} = props;
+  props = _props;
+  if (vanilla) {
+    return isHydrated ? null : React__namespace.createElement("script", _rollupPluginBabelHelpers["extends"]({}, props, {
+      src: manifest.entry.module,
+      type: "module"
+    }));
+  }
+
   let initialScripts = React__namespace.useMemo(() => {
     let contextScript = serverHandoffString ? `window.__remixContext = ${serverHandoffString};` : "";
     let routeModulesScript = `${matches.map((match, index) => `import * as route${index} from ${JSON.stringify(manifest.routes[match.route.id].module)};`).join("\n")}
diff --git a/node_modules/@remix-run/react/esm/components.js b/node_modules/@remix-run/react/esm/components.js
index 8bb3792..ae1627d 100644
--- a/node_modules/@remix-run/react/esm/components.js
+++ b/node_modules/@remix-run/react/esm/components.js
@@ -584,6 +584,16 @@ function Scripts(props) {
   React.useEffect(() => {
     isHydrated = true;
   }, []);
+
+  let {vanilla, ..._props} = props;
+  props = _props;
+  if (vanilla) {
+    return isHydrated ? null : React__namespace.createElement("script", _rollupPluginBabelHelpers["extends"]({}, props, {
+      src: manifest.entry.module,
+      type: "module"
+    }));
+  }
+
   let initialScripts = React.useMemo(() => {
     let contextScript = serverHandoffString ? `window.__remixContext = ${serverHandoffString};` : "";
     let routeModulesScript = `${matches.map((match, index) => `import * as route${index} from ${JSON.stringify(manifest.routes[match.route.id].module)};`).join("\n")}
